app/views/admin/articles/_individual_article.html.erb

```diff
@@ -40,18 +40,46 @@
       </a>
     <% end %>
 
-    <h2 class="crayons-subtitle-1">
-      <a href="<%= article.path %>" target="_blank" rel="noopener" class="c-link c-link--branded">
-        <%= article.title %>
-      </a>
-    </h2>
+    <div class="flex justify-between gap-4 items-center">
+      <h2 class="crayons-subtitle-1">
+        <a href="<%= article.path %>" target="_blank" rel="noopener" class="c-link c-link--branded">
+          <%= article.title %>
+        </a>
+      </h2>
+      <div class="flex items-center gap-1">
+        <% counts = Hash.new(0) %>
```

Comment:
Instead of building the data this way in the view, could we instead add a method in `article.rb`, something like:

```ruby
  def privileged_reaction_counts
    @privileged_reaction_counts ||= reactions.privileged_category.group(:category).count
  end
```

(Arguably, this would work even better in a decorator, but we don't have an admin-specific decorator for article, and I don't think it's worth it to create one _just_ for this, and I would not add this to a decorator that's otherwise _entirely_ for public views.)

Reply:
Thanks @jaw6 for the improvement. I applied this change but the only issue is that I had to make this method public, otherwise I was unable to access it. 
Is there a better way to use this method without making it public?

Reply:
It comes down to _where_ the method is defined in the file — [the `private` declaration is around line 600 in `article.rb`](https://github.com/forem/forem/blob/e34bec57833ec2c114e171584672872d987f9810/app/models/article.rb#L601) so the new method needs to be defined before that line, instead of after.

Reply:
Thanks for this clarification.